web/src/main/java/org/openmrs/web/filter/GZIPRequestWrapper.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.filter;
+
+import java.io.IOException;
+
+import javax.servlet.ServletInputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Wraps Request for GZipFilter
+ *
+ * @author Tomasz Mueller tmueller@soldevelo.com
+ */
+public class GZIPRequestWrapper extends HttpServletRequestWrapper {
+	
+	protected HttpServletRequest origRequest = null;
+	
+	protected ServletInputStream stream = null;
+	
+	protected int error = 0;
+	
+	public GZIPRequestWrapper(HttpServletRequest request) {
+		super(request);
+		origRequest = request;
+		
+	}
+	
+	public ServletInputStream createInputStream() throws IOException {
```

Comment:
Why don't you simply create the stream in constructor and get rid of this method?